### PR TITLE
constructing.responses.from.caches spec tests

### PR DESCRIPTION
**Pull request by akrambek**
Spec tests for https://tools.ietf.org/html/rfc7234#section-4

### CODE REVIEW

File: src/main/scripts/org/reaktivity/specification/nukleus/http_cache/streams/proxy/rfc7234/not.use.cache.that.varys.with.asterisk.value/connect/client.rpt [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+connected
+
+write close
+
+read nukleus:begin.ext ${http:header(":status", "200")}
+                       ${http:header("content-length", "11")}
+                       ${http:header("cache-control", "max-age=5")}
+                       ${http:header("date", date)}
+                       ${http:header("last-modified", date)}
```
Comment (dpwspoon) [inline src/main/scripts/org/reaktivity/specification/nukleus/http_cache/streams/proxy/rfc7234/not.use.cache.that.varys.with.asterisk.value/connect/client.rpt]: suggest not using last-modified (maybe in all tests in the pr

File: src/test/java/org/reaktivity/specification/nukleus/http_cache/streams/Rfc7234ProxyConnectIT.java [hunk 2]
```diff
@@ -428,4 +428,28 @@ public void shouldProxyResponseTooLargeToCache() throws Exception
         k3po.notifyBarrier("ROUTED_PROXY");
         k3po.finish();
     }
+
+    @Test
+    @Specification({
+            "${streams}/invalidate.multiple.max-age/connect/client",
+            "${streams}/invalidate.multiple.max-age/connect/server",
+    })
+    public void shouldNotCacheWithMultipleMaxAge() throws Exception
```
Comment (dpwspoon) [inline src/test/java/org/reaktivity/specification/nukleus/http_cache/streams/Rfc7234ProxyConnectIT.java]: I think you are missing test methods here

File: src/main/scripts/org/reaktivity/specification/nukleus/http_cache/streams/proxy/rfc7234/cache.with.freshened.response.that.updated.by.weak.validator/accept/client.rpt [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+                        ${http:header(":path", "/path")}
+                        ${http:header(":authority", "localhost:8080")}
+                        ${http:header("if-modified-since", date)}
+                        ${http:header("if-none-match", etag)}
+
+connected
+
+write close
+
+read nukleus:begin.ext ${http:header(":status", "304 (Not Modified)")}
```
Comment (dpwspoon) [inline src/main/scripts/org/reaktivity/specification/nukleus/http_cache/streams/proxy/rfc7234/cache.with.freshened.response.that.updated.by.weak.validator/accept/client.rpt]: I would expect the 304 to be returned from the cache/nukelus and not go through.  AM I missing something.

**Review by dpwspoon — CHANGES_REQUESTED** [review]

File: src/main/scripts/org/reaktivity/specification/nukleus/http_cache/streams/proxy/rfc7234/cache.with.freshened.response.that.updated.by.weak.validator/accept/client.rpt (hunk 3 above)
Comment (akrambek) [reply src/main/scripts/org/reaktivity/specification/nukleus/http_cache/streams/proxy/rfc7234/cache.with.freshened.response.that.updated.by.weak.validator/accept/client.rpt]: I was trying follow https://tools.ietf.org/html/rfc7234#section-4.3.4? Perhaps you can double confirm my understanding.

File: src/test/java/org/reaktivity/specification/nukleus/http_cache/streams/Rfc7234ProxyConnectIT.java (hunk 2 above)
Comment (akrambek) [reply src/test/java/org/reaktivity/specification/nukleus/http_cache/streams/Rfc7234ProxyConnectIT.java]: What do mean?

**Review by akrambek — COMMENTED** [review]
Updated comments

File: src/main/scripts/org/reaktivity/specification/nukleus/http_cache/streams/proxy/rfc7234/cache.with.freshened.response.that.updated.by.weak.validator/accept/client.rpt (hunk 3 above)
Comment (dpwspoon) [reply src/main/scripts/org/reaktivity/specification/nukleus/http_cache/streams/proxy/rfc7234/cache.with.freshened.response.that.updated.by.weak.validator/accept/client.rpt]: I don't think this test is needed, (at least in this incarnation).  The cache is refreshed in the cache, BUT I would still expect a 304 to be returned the client.  I don't see anything in section-4.3.4 to change that behavior on weak etag (again, please correct me if I'm missing something).

Comment (akrambek) [reply src/main/scripts/org/reaktivity/specification/nukleus/http_cache/streams/proxy/rfc7234/cache.with.freshened.response.that.updated.by.weak.validator/accept/client.rpt]: Yes, I recall our discussion about returning 304. I agree this test is not really needed.

File: src/main/scripts/org/reaktivity/specification/nukleus/http_cache/streams/proxy/rfc7234/expire.max-age/connect/server.rpt [hunk 4]
```diff
@@ -16,7 +16,7 @@
 
 property newProxyConnectRef ${nuklei:newReferenceId()} # external scope
 property date ${http_cache:date()} # shared property
-property date2 ${http_cache:datePlus(2)} # shared property
+
```
Comment (dpwspoon) [inline src/main/scripts/org/reaktivity/specification/nukleus/http_cache/streams/proxy/rfc7234/expire.max-age/connect/server.rpt]: date2 is used in this script down below

**Review by dpwspoon — CHANGES_REQUESTED** [review]

**Review by dpwspoon — APPROVED** [review]

**Pull request merged**